toolkit/tools/pkg/imagecustomizerlib/customizepackages.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+)
+
+func updatePackages(buildDir string, baseConfigPath string, packagesToAddLists []string, packagesToAdd []string,
+	packagesToRemoveLists []string, packagesToRemove []string, imageChroot *safechroot.Chroot, rpmsSources []string,
+	useBaseImageRpmRepos bool,
+) error {
+	var err error
+
+	allPackagesToRemove, err := collectPackagesList(baseConfigPath, packagesToRemoveLists, packagesToRemove)
+	if err != nil {
+		return err
+	}
+
+	allPackagesToAdd, err := collectPackagesList(baseConfigPath, packagesToAddLists, packagesToAdd)
+	if err != nil {
+		return err
+	}
+
+	err = removePackages(buildDir, allPackagesToRemove, imageChroot)
+	if err != nil {
+		return err
+	}
+
+	err = installPackages(buildDir, allPackagesToAdd, imageChroot, rpmsSources, useBaseImageRpmRepos)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func collectPackagesList(baseConfigPath string, packageLists []string, packages []string) ([]string, error) {
+	var err error
+
+	// Read in the packages from the package list files.
```

Comment:
In our main tools, we also consider the kernel package as a special case. Does there need to be any special handling for the kernel for post-customization? (I'm not actually sure there is a reason for special handling in the main tools anymore tbh).

Reply:
Do you know what the original reason for the special handling?

That being said, I do have another task to deal with the kernel package installation/replacement.

Reply:
No, I think that is probably lost to the mists of time. I'll ask around and see if anyone recalls.